fund.go

```diff
@@ -0,0 +1,293 @@
+	ClientId   float64
+	ResourceId float64
+}
+
+type FundDecrementRequest struct {
+	Id                string `json:"-"`
+	Locale            string `json:"locale"`
+	DateFormat        string `json:"dateFormat"`
+	TransactionDate   string `json:"transactionDate"`
+	TransactionAmount string `json:"transactionAmount"`
+	PaymentTypeId     string `json:"paymentTypeId"`
+}
+
+type FundDecrementResponse struct {
+	OfficeId   float64
+	ClientId   float64
+	ResourceId float64
+}
+
+type FundValueRequest struct {
+	Id string `json:"-"`
+}
+
+type Summary struct {
+	Amount           float64 `json:"accountBalance"`
+	TotalDeposits    float64 `json:"totalDeposits"`
+	TotalWithdrawals float64 `json:"totalWithdrawals"`
+}
+
+type FundValueResponse struct {
+	Id         int64   `json:"id"`
+	AccountNo  string  `json:"accountNo"`
+	ClientId   int64   `json:"clientId"`
+	ClientName string  `json:"clientName"`
+	Statement  Summary `json:"summary"`
+}
+
+type FundsRequest struct{}
+
+type FundsResponse struct {
+	TotalFilteredRecords int64               `json:"totalFilteredRecords"`
+	FundDetail           []FundValueResponse `json:"pageItems"`
+}
+
+type AuthenticationKey struct {
+	Data string `json:"base64EncodedAuthenticationKey"`
+}
+
+// TODO: The service will send username and password in headers
+func getAuthenticationKey() (string, error) {
+	if authenticationKey.Data != "" {
+		return authenticationKey.Data, nil
+	}
+	tempPath, _ := url.Parse("authentication?username=mifos&password=password")
+	tempPath1 := client.HostName.ResolveReference(tempPath).String()
+	path := strings.Replace(tempPath1, "/savingsaccounts", "", -1)
+	req, err := http.NewRequest("POST", path, bytes.NewBuffer([]byte{}))
+	if err != nil {
+		log.Println(err)
+		return "", err
+	}
+	req.Header.Set("Content-Type", "application/json")
+	req.Header.Set("fineract-platform-tenantid", "default")
+
+	var resp *http.Response
+	errTry := highbrow.Try(5, func() error {
+		resp, err = client.HttpClient.Do(req)
+		return err
+	})
+	if errTry != nil {
+		log.Println(errTry.Error())
+		rawMessage := json.RawMessage([]byte(errTry.Error()))
+		return "", &FineractError{ErrCodeSerialization, &rawMessage}
+	}
+	defer resp.Body.Close()
+
+	body, _ := ioutil.ReadAll(resp.Body)
+	if resp.StatusCode != 200 {
+		rawMessage := json.RawMessage(body)
+		return "", &FineractError{GetFineractStatusCode(resp.StatusCode), &rawMessage}
+	}
+
+	err = json.Unmarshal(body, &authenticationKey)
+	if err != nil {
+		rawMessage := json.RawMessage([]byte(err.Error()))
+		return "", &FineractError{ErrCodeSerialization, &rawMessage}
+	}
+	return authenticationKey.Data, err
+
+}
+
+func (client *Client) FundIncrement(request FundIncrementRequest) (*FundIncrementResponse, error) {
+	b, err := json.Marshal(request)
+	if err != nil {
+		log.Println(err)
+		return nil, err
+	}
+	tempPath, _ := url.Parse(path.Join(request.Id, "transactions?command=deposit"))
+	path := client.HostName.ResolveReference(tempPath).String()
```

Comment:
This is nearly identical code for all the methods. Please move this to a common method on *Client.